src/map/pc.h

```diff
@@ -173,6 +173,17 @@ struct pc_combos {
 	int id; /* this combo id */
 };
 
+/** Auto-cast related data. **/
+struct autocast_data {
+	enum autocast_type type; // The auto-cast type.
+	int skill_id; // The auto-cast skill ID.
+	int skill_lv; // The auto-cast skill level.
+	bool itemskill_conditions_checked; // Used by itemskill() script command, to prevent second check of conditions after target was selected.
```

Comment:
probably here better use bit fields for bools as it was for old fields?
this will reduce memory usage a bit

Reply:
Wouldn't make a difference in this case.
 * `bool` has a size of 1 byte which makes a total of 4 bytes.
 * A struct containing 4 `unsigned int` fields with a width of 1 (that's how we use bit fields) has a size of 4 bytes.
```
struct {
	unsigned int bit_field1 : 1;
	unsigned int bit_field2 : 1;
	unsigned int bit_field3 : 1;
	unsigned int bit_field4 : 1;
} bit_fields;
ShowError("sizeof(bit_fields): %d\n", sizeof(bit_fields));
struct {
	bool boolean1;
	bool boolean2;
	bool boolean3;
	bool boolean4;
} booleans;
ShowError("sizeof(booleans): %d\n", sizeof(booleans));
```
```
[Error]: sizeof(bit_fields): 4
[Error]: sizeof(booleans): 4
```

There may be compiler depending differences in fields sizes, but as far as i know, this is the most common variation.

Reply:
in this case yes, but if add one new flag, it may use 8 bytes for bools (because padding) and still 4 bytes for bit fields.
But yes probably left it as is for now, if add new flags, then may be convert to bit fields